src/components/NavBar/NavBar.jsx

```diff
@@ -80,8 +83,26 @@ const NavBar = () => {
               <Box>
                 <NavbarLinks />
               </Box>
+
               <Box sx={{ flexGrow: 1 }} />
+
+              {/* DESKTOP */}
```

Comment:
Generally I think it's better to put these kinds of things into the variable names themselves, rather than comments like this. That allows the language itself to validate and enforce this information. So like instead of doing this comment, have something like:

```
<Box sx={{ flexGrow: 1}}/>
  <DesktopLayout/>
  <MobileLayout/>
</Box>
```

Reply:
So, more of a self-documenting code?  How/When do you determine that there's too many child components for one parent?

Like the `Navbar` would have 6 files for it in that case.  Is that fine? Too much?   I'm definitely okay with pulling things out into their own component - should I just make those components in this file?

What are the best practices here?

Reply:
That's a very good question, and there's no hard and fast rule for this. One of the guides a lot of devs use is the [Single Responsibility Principle](https://stackify.com/solid-design-principles/). A single component should only be responsible for one or two things. If it gets beyond that we probably want to refactor. Currently this navbar has a few responsibilities: 
- checking if the user is logged in,
-  checking if it should render in mobile or desktop view
- actually rendering the mobile and desktop views 
- handling the logout modal.

That's a lot of things. If we can separate some of these responsibilities out into their own components, it can be easier to understand how everything works together.

For the "too many child components for one parent", you're right it sometimes makes sense to combine a couple things together. So like maybe this navbar's responsibility should be figuring out which navbar to display; the login navbar, the desktop navbar, or the mobile navbar. Then there's just some switch logic.

Another rule of thumb is how long a section is getting. Functions generally don't need to be much longer than a dozen or so lines, files don't need to be much longer than a couple hundred. The `NavBar.jsx` file is still only about 200 lines, so if you want to keep the definitions for the mobile layout and the desktop layout in the same file, I'm ok with that. 

Proper software design is a hotly debated topic and there are dozens of books on it.

---

Most IDEs also include automatic refactoring features. [VSCode's React Refactor plugin](https://marketplace.visualstudio.com/items?itemName=planbcoding.vscode-react-refactor) can help you create these components pretty quickly.

Reply:
Just found this, but haven't tested yet:
<img width="746" alt="Screenshot 2023-07-21 at 10 50 06 AM" src="https://github.com/codeforpdx/PASS/assets/71395931/2256f87d-a6c0-43de-8ef1-9fb26806ac89">

https://jestjs.io/docs/tutorial-react#dom-testing
down in the `__tests__/CheckboxWithLabel-test.js` section

Reply:
Thanks for all this @timbot1789 

---

I don't mind pulling out and making `NavbarLoggedOut`, `NavbarDesktop`, `NavbarMobile` files, and then rendering the correct version in `NavBar.jsx` if you feel that's the better approach for readability/maintainability.

Reply:
I think that's a good idea. I was having a hard time reading that file myself